src/rng.rs

```diff
@@ -10,10 +10,47 @@
 //! [`Rng`] trait
 
 use rand_core::{Error, RngCore};
-use crate::distributions::uniform::{SampleBorrow, SampleUniform, UniformSampler};
+use crate::distributions::uniform::{SampleUniform, UniformSampler};
 use crate::distributions::{self, Distribution, Standard};
 use core::num::Wrapping;
 use core::{mem, slice};
+use core::ops::{Range, RangeInclusive};
+
+/// Range that supports generating a single sample efficiently.
+///
+/// Any type implementing this trait can be used to specify the sampled range
+/// for `Rng::gen_range`.
+pub trait SampleRange<T> {
```

Comment:
I think this belongs in the distributions::uniform module?

Reply:
Done.